service.subtitles.napiprojekt/resources/lib/NapiProjekt.py

```diff
@@ -0,0 +1,150 @@
+from urllib import request, parse
+import xbmc
+import xbmcaddon
+
+import base64
+import traceback
+from xml.dom import minidom
+import requests
+
+__addon__ = xbmcaddon.Addon()
+__scriptname__ = __addon__.getAddonInfo('name')
+
+class NapiProjektHelper:
+    def __init__(self, filename, md5hash):
+        self.info = {}
+        self.filename = filename
+        self.url = "http://napiprojekt.pl/api/api-napiprojekt3.php"
+        self.md5hash = md5hash
+
+    def log(self, msg=None, ex=None):
+        if ex:
+            level = xbmc.LOGERROR
+            msg = traceback.format_exc()
+        else:
+            level = xbmc.LOGINFO
+
+        xbmc.log((u"### [%s] - %s" % (__scriptname__, msg)), level=level)
+
+    def notify(self, msg):
+        xbmc.executebuiltin((u'Notification(%s,%s)' % (__scriptname__, msg)))
+
+    def search(self, item, t):
+        subtitle_list = []
+
+        for language in item["3let_language"]:
+            language = "pl" if language == "pol" else language
+            params = {
+                "l": language.upper(),
+                "f": self.md5hash,
+                "t": t,
+                "v": "other",
+                "kolejka": "false",
+                "nick": "",
+                "pass": "",
+                "napios": "Linux"
+            }
+
+            self.log(params)
+
+            url = "http://napiprojekt.pl/unit_napisy/dl.php?" + urllib.parse.urlencode(params)
+            subs = urllib.request.urlopen(url).read()
+
+            print(subs)
+
+            if subs[0:3] != b'NPc':
+                subtitle_list.append({"language": language, "is_preferred": language == item["preferredlanguage"]})
+
+        return sorted(subtitle_list, key=lambda x: (x['is_preferred']), reverse=True)
+
+    def download(self, language="PL"):
+        values = {
+            "mode": "1",
+            # "client":"NapiTux",
+            "client": "NapiProjektPython",
+            "client_ver": "0.1",
+            "downloaded_subtitles_id": self.md5hash,
+            "downloaded_subtitles_txt": "1",
+            "downloaded_subtitles_lang": language
+        }
+
+        self.log(values)
+
+        try:
+            # data = parse.urlencode(values).encode()
+            response = requests.post(self.url, data=values).text
+
+            self.log(response)
+            DOMTree = minidom.parseString(response)
+
+            cNodes = DOMTree.childNodes
+            if cNodes[0].getElementsByTagName("status"):
+                text = base64.b64decode(
+                    cNodes[0].getElementsByTagName("subtitles")[0].getElementsByTagName("content")[0].childNodes[
+                        0].data)
+                filename2 = self.filename + ".txt"
+                print(self.filename)
+                print(filename2)
+                open(filename2, "wb").write(text)
+                return filename2
+
+        except Exception as e:
+            self.notify('Download error, check log')
```

Comment:
UI notifications/dialogs have to have the text localized. Please move the string to strings.po.

Reply:
How to get string from Strings.po in code?

Reply:
https://codedocs.xyz/xbmc/xbmc/group__python__xbmc.html#ga82834997987f068af982d91feb40bf9f

Have you tried to contact the original maintainer of this addon or have you tried to submit your fixes upstream?

Reply:
I have not contacted to him. Addon is not modified for couple years. Will try to contact with him.

Reply:
Right. You're free to takeover as long as there's proof the addon is unmaintained, you got authorization from the original maintainer or he/she is actually MIA for some time. The addon in fact looks unmaintained though.